plugins/autodim.koplugin/main.lua

```diff
@@ -0,0 +1,250 @@
+                            self:_schedule_autodim_task()
+                            if touchmenu_instance then touchmenu_instance:updateItems() end
+                        end,
+                        extra_text = _("Disable"),
+                        extra_callback = function()
+                            self.autodim_starttime_m = -1
+                            G_reader_settings:saveSetting("autodim_starttime_minutes", -1)
+                            self:_schedule_autodim_task()
+                            if touchmenu_instance then touchmenu_instance:updateItems() end
+                        end,
+                    }
+                    UIManager:show(idle_dialog)
+                    if touchmenu_instance then touchmenu_instance:updateItems() end
+                end,
+                keep_menu_open = true,
+            },
+            {
+                text_func = function()
+                    return T(_("Dimmer duration: %1 seconds"), self.autodim_duration_s)
+                end,
+                enabled_func = function() return self.autodim_starttime_m > 0 end,
+                callback = function(touchmenu_instance)
+                    local dimmer_dialog = SpinWidget:new{
+                        title_text = _("Automatic dimmer duration"),
+                        info_text = _("Time is in seconds."),
+                        value = self.autodim_duration_s,
+                        default_value = DEFAULT_AUTODIM_DURATION_S,
+                        value_min = 0,
+                        value_max = 300,
+                        value_step = 1,
+                        value_hold_step = 10,
+                        callback = function(spin)
+                            if not spin then return end
+                            self.autodim_duration_s = spin.value
+                            G_reader_settings:saveSetting("autodim_duration_seconds", spin.value)
+                            self:_schedule_autodim_task()
+                            if touchmenu_instance then touchmenu_instance:updateItems() end
+                        end,
+                    }
+                    UIManager:show(dimmer_dialog)
+                    if touchmenu_instance then touchmenu_instance:updateItems() end
+                end,
+                keep_menu_open = true,
+            },
+            {
+                text_func = function()
+                    return T(_("Dimming percentage: %1%"), self.autodim_endpercentage)
+                end,
+                enabled_func = function() return self.autodim_starttime_m > 0 end,
+                callback = function(touchmenu_instance)
+                    local percentage_dialog = SpinWidget:new{
+                        title_text = _("Percentage"),
+                        info_text = _("Enter the percentage of you want the frontlight to dim to. 50% means half of your normal intensity."),
```

Comment:
`Specify the amount of attenuation you want to apply to the brightness level. 50% will halve it.`

Reply:
Oops, nope. It's just the final value at the end of the ramp.

Reply:
Okay, reading the code helped. It's indeed an attenuation factor.

Reply:
(I would change the constant & setting names to make that clearer).

Reply:
Attenuation? Is that really preferable to something based on a word like decrease or dim? ;-)

Reply:
Nope, but I couldn't come up with something with `dim` that wasn't overly wordy ;p).

Reply:
Oh, those words. 
Attenuation factor might be misleading. Given the factor is 10%: Does this mean the end will be 10% _less than_ or 10% _of the_ start value?

I summarize the behavior maybe this helps in finding a suitable term.

The current behavior is:
The user selects his desired FL-intensity and reads. When the idle period is over, the current FL-intensity is stored. This will be 100%. Then the dimming ramp goes down to the `endPercentage`.

So maybe we should call it:
`Enter the end value of the frontlight intensity as a percentage of the normal brightness. 0% means off.`

Reply:
So just reduce that to `A percentage of the normal brightness.` Lose the baggage.